packages/investor-yearn/src/YearnOpportunity.ts

```diff
@@ -0,0 +1,351 @@
+  fast: 1,
+  average: 0.8,
+  slow: 0.5
+})
+
+export class YearnOpportunity
+  implements
+    InvestorOpportunity<PreparedTransaction, VaultMetadata>,
+    ApprovalRequired<PreparedTransaction>
+{
+  readonly #internals: {
+    chainAdapter: ChainAdapter<ChainTypes.Ethereum>
+    dryRun?: true
+    logger?: Logger
+    routerContract: Contract
+    vault: Vault
+    web3: Web3
+    yearn: Yearn<ChainId>
+  }
+  public readonly apy: BigNumber
+  public readonly displayName: string
+  public readonly version: string
+  public readonly symbol: string
+  public readonly name: string
+  public readonly id: string
+  public readonly isApprovalRequired: true
+  public readonly isNew: boolean
+  public readonly metadata: VaultMetadata
+  public readonly underlyingAsset: { assetId: string; balance: BigNumber }
+  public readonly positionAsset: {
+    assetId: string
+    balance: BigNumber
+    underlyingPerPosition: BigNumber
+  }
+  public readonly feeAsset: { assetId: string }
+  public readonly price: BigNumber
+  public readonly supply: BigNumber
+  public readonly tvl: {
+    assetId: string
+    balance: BigNumber
+    balanceUsdc: BigNumber
+  }
+
+  constructor(deps: YearnOpportunityDeps, vault: Vault) {
+    this.#internals = {
+      chainAdapter: deps.chainAdapter,
+      dryRun: deps.dryRun,
+      logger: deps.logger ?? new Logger({ name: 'Yearn Opportunity' }),
+      routerContract: deps.contract,
+      vault,
+      web3: deps.web3,
+      yearn: deps.yearnSdk
+    }
+
+    this.id = vault.address
+    this.metadata = vault.metadata
+    this.displayName = vault.metadata.displayName || vault.name
+    this.version = vault.version
+    this.symbol = vault.symbol
+    this.apy = bnOrZero(vault.metadata.apy?.net_apy)
+    this.isNew = vault.metadata.apy?.type === 'new'
+    this.name = `${vault.metadata.displayName} ${vault.version}`
+    this.symbol = vault.metadata.symbol
+    // @TODO TotalSupply from the API awas 0
+    this.supply = bnOrZero(vault.metadata.totalSupply)
+    this.tvl = {
+      balance: bnOrZero(vault.underlyingTokenBalance.amount),
+      balanceUsdc: bnOrZero(vault.underlyingTokenBalance.amountUsdc),
+      assetId: toAssetId({
+        chainId: 'eip155:1',
+        assetNamespace: 'erc20',
+        assetReference: vault.tokenId || vault.token
+      })
+    }
+    this.underlyingAsset = {
+      balance: bnOrZero(0),
+      assetId: toAssetId({
+        chainId: 'eip155:1',
+        assetNamespace: 'erc20',
+        assetReference: vault.tokenId || vault.token
+      })
+    }
+    this.positionAsset = {
+      balance: bnOrZero(0),
+      assetId: toAssetId({
+        chainId: 'eip155:1',
+        assetNamespace: 'erc20',
+        assetReference: this.id
+      }),
+      underlyingPerPosition: bnOrZero(vault.metadata.pricePerShare)
+    }
+    this.feeAsset = {
+      assetId: 'eip155:1/slip44:60'
+    }
+  }
+
+  // TODO(theobold): think through this function to see if we want to call this after yearn
```

Comment:
given we're writing brand new code, i'd prefer if we consider if we need this or not now.

given this class member is private, and seemingly unused, can we simply remove it?

Reply:
We left it in as a reminder for us to circle back to the question of whether we need this type of function or not, but I'm okay with removing it until we do.